app/models/spree/invoice.rb

```diff
@@ -0,0 +1,128 @@
+class Spree::Invoice < ActiveRecord::Base
+  belongs_to :invoiceable, polymorphic: true
+  validates :invoiceable, presence: true
+
+  before_save :set_date
+  before_save :set_number, if: :use_sequential_number?
+  after_save :increase_invoice_number, if: :use_sequential_number?
```

Comment:
I want to set the number before saving, but still have the number available if the save fails. See the complementary changes in `lib/spree_print_invoice/print_invoice_settings.rb`. :)